src/event/pmix_event_registration.c

```diff
@@ -382,30 +489,164 @@ static void reg_event_hdlr(int sd, short args, void *cbdata)
+        ev = (pmix_event_hdlr_t*)pmix_list_get_last(cd->list);
+        if (PMIX_EVENT_ORDER_LAST == ev->precedence) {
+            /* this is an error */
+            --pmix_globals.events.nhdlrs;
+            rc = PMIX_ERR_EVENT_REGISTRATION;
+            index = UINT_MAX;
+            PMIX_RELEASE(evhdlr);
+            goto ack;
+        }
+        /* append it to the list */
+        pmix_list_append(cd->list, &evhdlr->super);
+    } else if (PMIX_EVENT_ORDER_PREPEND == location) {
+        /* we know the list isn't empty - check the first element to see if
+         * it is designated to be "first". If so, then we need to put this
+         * right after it */
+        ev = (pmix_event_hdlr_t*)pmix_list_get_first(cd->list);
+        if (PMIX_EVENT_ORDER_FIRST == ev->precedence) {
+            ev = (pmix_event_hdlr_t*)pmix_list_get_next(&ev->super);
+            if (NULL != ev) {
+                pmix_list_insert_pos(cd->list, &ev->super, &evhdlr->super);
+            } else {
+                /* we are at the end of the list */
+                pmix_list_append(cd->list, &evhdlr->super);
+            }
+        } else {
+            pmix_list_prepend(cd->list, &evhdlr->super);
+        }
+    } else if (PMIX_EVENT_ORDER_APPEND == location) {
+        /* we know the list isn't empty - check the last element to see if
+         * it is designated to be "last". If so, then we need to put this
+         * right before it */
+        ev = (pmix_event_hdlr_t*)pmix_list_get_last(cd->list);
+        if (PMIX_EVENT_ORDER_LAST == ev->precedence) {
+            pmix_list_insert_pos(cd->list, &ev->super, &evhdlr->super);
+        } else {
+            pmix_list_append(cd->list, &evhdlr->super);
+        }
+    } else {
+        /* find the named event */
+        found = false;
+        PMIX_LIST_FOREACH(ev, cd->list, pmix_event_hdlr_t) {
+            if (NULL == ev->name) {
+                continue;
+            }
+            if (0 == strcmp(ev->name, name)) {
+               if (PMIX_EVENT_ORDER_BEFORE == location) {
+                    /* put it before this handler */
+                    pmix_list_insert_pos(cd->list, &ev->super, &evhdlr->super);
+               } else {
+                   /* put it after this handler */
+                    ev = (pmix_event_hdlr_t*)pmix_list_get_next(&ev->super);
+                    if (NULL != ev) {
+                        pmix_list_insert_pos(cd->list, &ev->super, &evhdlr->super);
+                    } else {
+                        /* we are at the end of the list */
+                        pmix_list_append(cd->list, &evhdlr->super);
+                    }
+               }
+               found = true;
+               break;
+            }
+        }
+        /* if the handler wasn't found, then it may show up later - so
+         * for now just prepend it to the list */
+        if (!found) {
+            pmix_list_prepend(cd->list, &evhdlr->super);
```

Comment:
Maybe I missed it, but I did not see the piece of code to reorder the list when inserting the BEFORE-AFTER event